### PR TITLE
 sanic: 21.3.4 -> 21.9.1,   sanic-routing: 0.6.2 -> 0.7.2

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

ZHF: https://github.com/NixOS/nixpkgs/issues/144627

- Mark `pytest-sanic` as broken. It doesn't build with newer versions of `sanic`, since `sanic.websocket` was removed.
- Disable some tests that don't work. ~~Most of them are related to the lack of `sanic` on `PATH`, so it should be safe.~~

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by thiagokokada — COMMENTED** [review]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.pytest-sanic</li>
    <li>python39Packages.pytest-sanic</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.json-logging</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.sanic-routing</li>
    <li>python38Packages.sanic-testing</li>
    <li>python39Packages.entrance</li>
    <li>python39Packages.entrance-with-router-features</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.sanic</li>
    <li>python39Packages.sanic-auth</li>
    <li>python39Packages.sanic-routing</li>
    <li>python39Packages.sanic-testing</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytest-sanic/default.nix [hunk 1]
```diff
@@ -47,5 +47,6 @@ buildPythonPackage rec {
     homepage = "https://github.com/yunstanford/pytest-sanic/";
     license = licenses.asl20;
     maintainers = with maintainers; [ costrouc ];
+    broken = true; # 2021-11-04
```
Comment (thiagokokada) [inline pkgs/development/python-modules/pytest-sanic/default.nix:50]: This package is not dependency of any other package now that `sanic` itself doesn't depend on it.

It is a good candidate for removal.

**Comment by risicle** [comment]
`sanic` fails the test `test_no_exceptions_when_cancel_pending_request` on macos 10.15 with:

```
obj = <Process name='Process-1' parent=68540 initial>
file = <_io.BytesIO object at 0x112e7e400>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'test_no_exceptions_when_cancel_pending_request.<locals>.ping'
```

which looks awfully similar to the problems in https://github.com/sanic-org/sanic/issues/1774

File: pkgs/development/python-modules/sanic/default.nix [hunk 2]
```diff
@@ -23,28 +22,19 @@
...
-    })
-  ];
-
   postPatch = ''
     # Loosen dependency requirements.
     substituteInPlace setup.py \
-      --replace '"pytest==5.2.1"' '"pytest"' \
+      --replace '"pytest==6.2.5"' '"pytest"' \
       --replace '"gunicorn==20.0.4"' '"gunicorn"' \
       --replace '"pytest-sanic",' "" \
```
Comment (thiagokokada) [inline pkgs/development/python-modules/sanic/default.nix:39]: While `sanic` still declares `pytest-sanic` as dependencies, it doesn't use it for anything as it is clear that the removal also doesn't broke anything.

**Comment by thiagokokada** [comment]
> `sanic` fails the test `test_no_exceptions_when_cancel_pending_request` on macos 10.15 with:
> 
> ```
> obj = <Process name='Process-1' parent=68540 initial>
> file = <_io.BytesIO object at 0x112e7e400>, protocol = None
> 
>     def dump(obj, file, protocol=None):
>         '''Replacement for pickle.dump() using ForkingPickler.'''
> >       ForkingPickler(file, protocol).dump(obj)
> E       AttributeError: Can't pickle local object 'test_no_exceptions_when_cancel_pending_request.<locals>.ping'
> ```
> 
> which looks awfully similar to the problems in [sanic-org/sanic#1774](https://github.com/sanic-org/sanic/issues/1774)

At least for me many other sanic tests fails on macOS, so I don't think it is worth to trying to make it work.

It already doesn't work currently.

**Comment by thiagokokada** [comment]
> > `sanic` fails the test `test_no_exceptions_when_cancel_pending_request` on macos 10.15 with:
> > ```
> > obj = <Process name='Process-1' parent=68540 initial>
> > file = <_io.BytesIO object at 0x112e7e400>, protocol = None
> > 
> >     def dump(obj, file, protocol=None):
> >         '''Replacement for pickle.dump() using ForkingPickler.'''
> > >       ForkingPickler(file, protocol).dump(obj)
> > E       AttributeError: Can't pickle local object 'test_no_exceptions_when_cancel_pending_request.<locals>.ping'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > which looks awfully similar to the problems in [sanic-org/sanic#1774](https://github.com/sanic-org/sanic/issues/1774)
> 
> At least for me many other sanic tests fails on macOS, so I don't think it is worth to trying to make it work.
> 
> It already doesn't work currently.

I could maybe just disable tests on macOS entirely. WDYT?

**Comment by risicle** [comment]
Am opening an upstream issue.

**Comment by thiagokokada** [comment]
> Am opening an upstream issue.

Could we merge this as-is? We can always fix darwin later (since it is already broken anyway).

**Comment by risicle** [comment]
The other sanic tests that fail on macos, for me at least, are due to attempting to build the python 3.8 and 3.9 versions at the same time, and with the lack of functioning sandboxing on darwin, compete for the same local port.

**Comment by thiagokokada** [comment]
> The other sanic tests that fail on macos, for me at least, are due to attempting to build the python 3.8 and 3.9 versions at the same time, and with the lack of functioning sandboxing on darwin, compete for the same local port.

Makes sense. I will just disable the failing tests on macOS them.

**Comment by risicle** [comment]
I think simply disabling that test on darwin will unblock us.

**Comment by risicle** [comment]
:point_up: i.e. that _single_ test.

**Comment by thiagokokada** [comment]
@risicle Done.

**Comment by risicle** [comment]
https://github.com/sanic-org/sanic/issues/2298

**Comment by thiagokokada** [comment]
> [sanic-org/sanic#2298](https://github.com/sanic-org/sanic/issues/2298)

Nice. Add issue link as comment in derivation.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15

File: pkgs/development/python-modules/sanic/default.nix [hunk 3]
```diff
@@ -23,28 +23,19 @@
...
     owner = "sanic-org";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0vldlic8gqcf56fqb31igycqf11syd9csk66v34w6dim54lcny2b";
+    sha256 = "sha256-TRrJr/L8AXLAARPjhBi2FxNh+jvxxdeMN24cT1njmqY=";
   };
 
-  patches = [
-    # Allow later websockets release, https://github.com/sanic-org/sanic/pull/2154
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sanic/default.nix:37]: Please remove the unused fetchpatch from inputs.

Comment (thiagokokada) [reply pkgs/development/python-modules/sanic/default.nix:37]: Will do any cleanups in this PR: https://github.com/NixOS/nixpkgs/pull/145022

**Pull request merged**